File: server/models/post.js
import mongoose from "mongoose";

const postSchema = new mongoose.Schema({
    description : {type : String , default :"" },
    media : {type : String , required : true},
    likes : {type: [String] , default : []},
    owner : {type : mongoose.Schema.Types.ObjectId , ref : "User" ,required : true },
    createdAt : {type : Date , default : new Date()}   ,
    comments : {type : [mongoose.Schema.Types.ObjectId] ,ref : 'Comment' }


})

export default mongoose.model('Post',postSchema )

File: Pinterest/app/src/main/java/com/hamza/pinterest/Models/Post.java
package com.hamza.pinterest.Models;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Post implements Serializable {

        String _id , media ,owner, description ;
        List<String> likes = new ArrayList<>() ;
        Date createdAt ;

        ArrayList<Comment> comments = new ArrayList<>();

    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public String getMedia() {
        return media;
    }

    public void setMedia(String media) {
        this.media = media;
    }

    public String getOwner() {
        return owner;
    }

    public void setOwner(String owner) {
        this.owner = owner;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<String> getLikes() {
        return likes;
    }

    public void setLikes(List<String> likes) {
        this.likes = likes;
    }

    public ArrayList<Comment> getCommentList() {
        return comments;
    }

    public void setCommentList(ArrayList<Comment> commentList) {
        this.comments = commentList;
    }
}
